### PR TITLE
BaSP - TG-12 - time sheets: get, get by id and create endpoints

### PR DESCRIPTION
**Pull request by anapaulib**

### CODE REVIEW

File: src/controllers/time-sheets.js [hunk 1]
```diff
@@ -2,22 +2,53 @@ import TimeSheets from '../models/Time-sheets';
...
+    return res.status(404).json({
+      message: 'An error has ocurred',
+      error: err,
+    });
+  }
+};
+
+export const getTimeSheetById = async (req, res) => {
+  try {
+    const timeSheet = await TimeSheets.findById(req.params.id);
```
[the remark this answers is not in the document]
Comment (anapaulib) [reply src/controllers/time-sheets.js:21]: Ya lo corregí, ahora lo pusheo.

File: src/controllers/time-sheets.js [hunk 2]
```diff
@@ -2,22 +2,59 @@ import TimeSheets from '../models/Time-sheets';
 
 export const getAllTimeSheets = async (req, res) => {
   try {
-    const timeSheets = await TimeSheets.find(req.query);
-    if (!timeSheets.lenght) {
-      return res.status(404).json({
-        message: 'Time sheet not found',
-        data: timeSheets,
+    const timeSheets = await TimeSheets.find();
```
Comment (lmalvasia) [inline src/controllers/time-sheets.js:5]:
```suggestion
    const timeSheets = await TimeSheets.find(req.query);
```

File: src/controllers/time-sheets.js [hunk 3]
```diff
@@ -2,22 +2,59 @@ import TimeSheets from '../models/Time-sheets';
...
-      message: 'Time sheet found',
-      data: timeSheets,
+    return res.status(404).json({
+      message: 'Time sheet not found',
+      error: true,
+    });
+  } catch (err) {
+    return res.status(404).json({
+      message: 'An error has occured',
+      error: err,
```
Comment (lmalvasia) [inline src/controllers/time-sheets.js:36]: The status should be 400.
The variable `error` should be a boolean value.
           return res.status(400).json({
               message: error, // error.toString()
               data: undefined,
               error: true
           });

File: src/controllers/time-sheets.js [hunk 4]
```diff
@@ -2,22 +2,59 @@ import TimeSheets from '../models/Time-sheets';
...
+      message: 'Time sheet created successfully',
+      data: result,
       error: false,
     });
   } catch (err) {
-    return res.json({
-      message: 'an error ocurred',
+    return res.status(404).json({
+      message: 'Time sheet cannot be created',
       error: err,
```
Comment (lmalvasia) [inline src/controllers/time-sheets.js:58]: The status should be 400.
The variable error should be a boolean value.
return res.status(400).json({
message: error, // error.toString()
data: undefined,
error: true
});

**Review by GinaSchP — APPROVED** [review]

**Review by lmalvasia — APPROVED** [review]

**Review by GinaSchP — APPROVED** [review]

**Pull request merged**